csrc/kernel_cache.cpp

```diff
@@ -1453,11 +1455,11 @@ std::unordered_map<Val*, const PolymorphicValue*> FusionKernelRuntime::
     std::vector<at::Tensor> group_runtime_outputs =
         runKernelWithInput(group_runtime_inputs, group_to_run);
     args_manager.updateWithSegmentOutputs(
-        group_to_run->outputs(), group_runtime_outputs, group_id);
+        group_to_run->outputs(), group_runtime_outputs, run_order_id);
     num_live_args_after_segment_runs_.push_back((int64_t)args.size());
 
     if (compute_overall_bw) {
-      const auto& executor = executors_.at(group_id);
+      const auto& executor = executors_.at(group_to_run->groupId());
```

Comment:
Sorry I am a little bit slow today, but why would `run_order_id` not be equal to `group-to_run->groupId()`? I thought that was how groupId was defined.

Reply:
groupId is set [here](https://github.com/NVIDIA/Fuser/blob/6c066c0c1f01ae984bdb6adf51f5edc4c529c023/csrc/fusion_segmenter.cpp#L4308). `groups()` is not sorted in run order.  

(I did verify this by putting a `NVF_ERROR(group_to_run->groupId() == run_order_id)` and saw many tests fail.)